src/components/base/Base.js

```diff
@@ -2194,8 +2194,23 @@ export default class BaseComponent {
     if (this.viewOnly) {
       this.updateViewOnlyValue(newValue);
     }
-
     this.updateOnChange(flags, changed);
+    this.root.components.forEach((c) => {
```

Comment:
This is not triggering `updateValue()`, it is revalidating `required` and adding the asterisk to the label if it is reevaluated to be required. In our testing did not seem to trigger an `updateValue()`, we just loop through the components and recalculate `required`.
Formio does not give you an option to update custom required / validation on change of any other component 

e.g.:
    `textfield1` => I want this to be required ONLY when `foo()` returns `true`
    `foo()` returns true when `textfield2` has the value `"x"`
    when `textfield2` changes to `"x"`, `textfield1` its `required` should get set to `true` without user interaction 

Using custom code, you cannot assume which fields will interact with other fields in regards to `required` because custom javascript execution is entirely up to the person building the form.

`Validator.check()` can only set `required` in one direction from false to true, which is why we were forced to set `required` to false before revalidating for fields that had custom required logic.